Firestore/Protos/build_protos.py

```diff
@@ -188,6 +194,36 @@ def __stub_non_buildable_files(self, out_dir):
     ])
 
 
+class CppProtobufGenerator(object):
+  """Runs protoc for C++ libprotobuf (used in testing)."""
+
+  def __init__(self, args, proto_files):
+    self.args = args
+    self.proto_files = proto_files
+
+  def run(self):
+    out_dir = os.path.join(self.args.output_dir, 'cpp')
+    mkdir(out_dir)
+
+    self.__run_generator(out_dir)
+
+    sources = collect_files(out_dir, '.pb.h', '.pb.cc')
+    # TODO(wilhuff): strip trailing whitespace?
```

Comment:
It's not something I care about. IMO, the fewer post process steps applied to generated code, the better.